File: 3.cpp
auto contains_duplicates(std::string s) -> bool
{
    std::vector<char> previous{};
    for (const auto& x : s)
    {
        if (std::find(previous.begin(), previous.end(), x) == std::end(previous))
        {
            previous.push_back(x);
        }
        else
        {
            return true;
        }
    }
    return false;
}

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if (s.size() == 0)
        {
            return 0;
        }
        std::string longest_substring = std::string{ s[0] };
        for (int i = 0; i <= s.size(); i++)
        {
            for (int j = 0; j <= s.size(); j++)
            {
                if (!contains_duplicates(s.substr(i, j)) && s.substr(i, j).size() > longest_substring.size())
                {
                    longest_substring = s.substr(i, j);
                }
            }
        }
        if (longest_substring == std::string{ s[0] } &&
            std::find(s.begin() + 1, s.end(), s[0]) == std::end(s))
        {
            return s.size();
        }
        else
        {
            return longest_substring.size();
        }
    }
};

File: 3.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        max_len = 0
        if len(s) == 0: return 0
        for l in range(0, len(s)):
            curr = ""
            for r in range(l, len(s)):
                if s[r] not in curr:
                    curr += str(s[r])
                    max_len = max(max_len, len(curr))
                else:
                    break
        return max_len


        
